tools/src/bin/test-index.rs

```diff
@@ -0,0 +1,176 @@
+use std::{env, str};
```

Comment:
This is great.  The extra build pass was annoying and we already had so much custom stuff going on, especially with potentially depending on running web-servers, that trying to fit into the cargo test infra wasn't really helping.  Thanks!